acceptance-tests/src/acceptance-test/java/tech/pegasys/teku/test/acceptance/DoppelgangerDetectorAcceptanceTest.java

```diff
@@ -107,11 +107,84 @@ void shouldDetectDoppelgangerAndAlert() throws Exception {
 
     secondValidatorClient.waitForLogMessageContaining("Validator doppelganger detected...");
     secondValidatorClient.waitForLogMessageContaining("Detected 8 validators doppelganger");
+    secondValidatorClient.waitForLogMessageContaining("Doppelganger detection check finished");
+    secondValidatorClient.waitForLogMessageContaining("Detected 8 validators doppelganger");
```

Comment:
why this one is checked again?
why it's not shutdown in this case (or no check)

Reply:
In the first iteration the behaviour was to stop the DG detection whenever we detect at least one active key but now it continues running until:
All the keys are detected as active OR it has been running for 2 epochs.

Keep in mind that there are two DG detection actions: 1. Shutdown (Triggered at the VC startup) 2. Alert (Triggered when adding keys via the Key-Manager API)
In the first case (shutdown at VC startup) it could makes sense to stop the DG check at the first DG detection, but that means you're not sure it the other keys are active or not.
In the second case (alert, Key-Manager API), the detected active keys are ignored and the other ones are imported. If we stop the DG at the first active detection that means we could import keys that could be active.
That's why we implemented this change: Do not stop after the first detection, keep running for at least 2 epochs...

A log is printed whenever a DG is detected AND another log is printed at the end of the DG check with a full report of ALL the detected DGs. In this acceptance test, there are 2 DG detected at the first round: 1. First log when they're first detected 2. Second log when the check is finished

Reply:
Getting it. So we confirmed detection on the 2nd epoch, what's next, we are going to shutdown? If yes, we should check it

Reply:
In this use case (Loading keys via the Key Manager API) there should be an alert and the keys are ignored (not imported). I added a check for that too (`Detected 8 validators that appear to be already active. The following keys have been ignored`). Good catch